src/commands/detectorDiagnostics.ts

```diff
@@ -0,0 +1,152 @@
+import { IActionContext } from "vscode-azureextensionui";
+import { AppLensARMresponse } from './models/detectorAppLens';
+import * as path from 'path';
+
+const util = require('util');
+const exec = util.promisify(require('child_process').exec);
+
+export default async function detectorDiagnostics(
+    context: IActionContext,
+    target: any
+  ): Promise<void> {
+    const cloudExplorer = await k8s.extension.cloudExplorer.v1;
+
+    if (cloudExplorer.available) {
+      const cloudTarget = await cloudExplorer.api.resolveCommandTarget(target);
+
+      if (cloudTarget && cloudTarget.cloudName === "Azure" &&
+            cloudTarget.nodeType === "resource" && cloudTarget.cloudResource.nodeType === "cluster") {
+            const armId = cloudTarget.cloudResource.armId;
+            // Get applens data
+            const clusterAppLensData = await getAppLensDetectorData(armId);
+            // Create webview
+            await CreateDetectorWebView(target.value.name, clusterAppLensData);
+        } else {
+          vscode.window.showInformationMessage('Please select kubernetes cluster.');
+        }
+    }
+}
+
+async function CreateDetectorWebView(
+  clusterName: string,
+  clusterAppLensData: (AppLensARMresponse | void)) {
+
+  if (clusterAppLensData) {
+    vscode.window.showInformationMessage("Loading...");
+    // Create webview
+    const panel = vscode.window.createWebviewPanel(
+      'AKS Diagnostics',
+      'AKS diagnostics view for: ' + clusterName,
+      vscode.ViewColumn.One,
+      {
+        enableScripts: true,
+        enableCommandUris: true
+      }
+    );
+
+    // Build webview from data.
+    panel.webview.html = getWebviewContent(clusterAppLensData);
+    vscode.window.showInformationMessage("AKS Diagnostics Completed.");
+  } else {
+    vscode.window.showInformationMessage('Selected Cluster has no data returned.');
+  }
+}
+
+async function getAppLensDetectorData(clusterARMId: string): Promise<AppLensARMresponse | void> {
+  // ARM api call.
+  const appLensARMAPI = `az rest -m GET -u "${clusterARMId}/detectors/mcrEndpointUpdate?api-version=2019-08-01&executeChildren=true"`;
+  const clusterAppLensData = await runAksShellCommand(appLensARMAPI);
+
+  return clusterAppLensData;
+}
+
+async function runAksShellCommand(azcomand: string): Promise<AppLensARMresponse | void> {
+  try {
+    const { stdout } = await exec(azcomand);
+    const appLensARMresponse = new AppLensARMresponse(JSON.parse(stdout));
+
+    return appLensARMresponse;
+  }catch (err) {
+    vscode.window.showInformationMessage('Error: ' + err);
+  }
+}
+
+function getWebviewContent(clusterdata: AppLensARMresponse) {
+    const webviewClusterData = clusterdata?.properties;
+    const vscodeExtensionPath = process.env.VSCODE_CWD;
+
+    if (vscodeExtensionPath === undefined) {
+      vscode.window.showErrorMessage("No Extension path");
+      // TODO: context.extensionPath is missing at command level.
+      return "CSS file path missing";
```

Comment:
You can't return a string that might be a message or file path with no indicator as to which it is!  Use a DU such as `Errorable`, or just throw here because it's totally unexpected.